Makefile

```diff
@@ -7,195 +7,166 @@ PG_VERSIONS?=12 11
-	docker push $(TIMESCALEDB_RELEASE_URL)-pg$(PG_MAJOR)$(POSTFIX) \
-	&& docker push $(TIMESCALEDB_LATEST_URL)-pg$(PG_MAJOR)$(POSTFIX)
-
-.PHONY: push-all
-push-all: push push-oss
+	docker stop $(DOCKER_TAG_PREPARE) || true
+	docker run -d --rm --name $(DOCKER_TAG_PREPARE) -e PGDATA=/tmp/pgdata --user=postgres $(DOCKER_TAG_PREPARE) \
+		sh -c 'initdb && timeout 60 postgres'
+	$(DOCKER_EXEC_COMMAND) sh -c 'while ! pg_isready; do sleep 1; done'
+	cat scripts/install_extensions.sql | $(DOCKER_EXEC_COMMAND) psql -AtXq  --set ON_ERROR_STOP=1
+	cat scripts/version_info.sql | $(DOCKER_EXEC_COMMAND) psql -AtXq > $(VAR_VERSION_INFO)
+	docker stop $(DOCKER_TAG_PREPARE) || true
+	if [ ! -z "$(TIMESCALE_TSDB_ADMIN)" -a "$(POSTFIX)" != "-oss" ]; then echo "tsdb_admin.version=$(TIMESCALE_TSDB_ADMIN)" >> $(VAR_VERSION_INFO); fi
+
+build: $(VAR_VERSION_INFO)
+	echo "FROM $(DOCKER_TAG_PREPARE)" | docker build --tag "$(DOCKER_TAG_LABELED)" - \
+	  $$(awk -F '=' '{printf "--label com.timescaledb.image."$$1"="$$2" "}' $(VAR_VERSION_INFO))
+
+build-oss: DOCKER_EXTRA_BUILDARGS= --build-arg OSS_ONLY=" -DAPACHE_ONLY=1"
+build-oss: DOCKER_TAG_POSTFIX=-oss
+build-oss: build
 
 # The purpose of publishing the images under many tags, is to provide
 # some choice to the user as to their appetite for volatility.
 #
-#  1. timescaledev/timescaledb-ha:pg12-latest
-#  2. timescaledev/timescaledb-ha:pg12-ts1.7-latest
-#  3. timescaledev/timescaledb-ha:pg12.3-ts1.7-latest
-#  4. timescaledev/timescaledb-ha:pg12.3-ts1.7.1-latest
-#  5. timescaledev/timescaledb-ha:pg12.3-ts1.7.1-pN
+#  1. timescale/timescaledb-ha:pg12-latest
+#  2. timescale/timescaledb-ha:pg12-ts1.7-latest
+#  3. timescale/timescaledb-ha:pg12.3-ts1.7-latest
+#  4. timescale/timescaledb-ha:pg12.3-ts1.7.1-latest
+#  5. timescale/timescaledb-ha:pg12.3-ts1.7.1-pN
 #
 # Tag 5 is immutable, and for every time we publish that image, we increase N by 1,
 # we start with N=0
-#
-# Our method of finding a patch version is quite brute force (`docker pull image`),
-# however, this should not happen that often.
-.PHONY: publish publish-oss
-publish publish-oss:
-	export PGVERSION=$$(docker inspect $(CICDIMAGE) | jq '.[0]."ContainerConfig"."Labels"."com.timescaledb.image.postgresql.version"' -r) \
-	&& export TSPATCH=$$(docker inspect $(CICDIMAGE) | jq '.[0]."ContainerConfig"."Labels"."com.timescaledb.image.timescaledb.version"' -r) \
-	&& export TSMINOR=$${TSPATCH%.*} \
-	&& for variant in pg$(PG_MAJOR)$(POSTFIX)-latest pg$(PG_MAJOR)-ts$${TSMINOR}$(POSTFIX)-latest pg$${PGVERSION}-ts$${TSMINOR}$(POSTFIX)-latest pg$${PGVERSION}-ts$${TSPATCH}$(POSTFIX)-latest; \
-		do \
-			docker tag $(CICDIMAGE) $(PUBLISH_REPOSITORY):$${variant} \
-			&& docker push $(PUBLISH_REPOSITORY):$${variant} || exit 1; \
+# Our method of finding a patch version is quite brute force (`docker pull image`), 
+# however, we should not be publishing that often, so we take the hit for now.
+publish: publish-mutable publish-next-patch-version
```

Comment:
Out of date comment, we no longer pull, we do http calls checking for `200` or `404`